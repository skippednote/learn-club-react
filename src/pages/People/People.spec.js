import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { People } from './index';
import fixture from './People.fixture';

const props = {
  people: fixture,
  error: null,
  loading: false,
  dispatch: jest.fn()
};

describe('People', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<People {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders a loading message when requesting data', () => {
    const props = {
      people: [],
      error: null,
      loading: true,
      dispatch: jest.fn()
    };
    const wrapper = shallow(<People {...props} />);
    const loading = wrapper.find('.loading');
    expect(loading).toHaveLength(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders a error message when request fails', () => {
    const props = {
      articles: [],
      error: { message: 'Request failed with status code 404' },
      loading: false,
      dispatch: jest.fn()
    };

    const wrapper = shallow(<People {...props} />);
    const loading = wrapper.find('.error');
    expect(loading).toHaveLength(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders PeopleList component', () => {
    const wrapper = shallow(<People {...props} />);
    const counter = wrapper.find('PeopleList');
    expect(counter).toHaveLength(1);
  });
});
