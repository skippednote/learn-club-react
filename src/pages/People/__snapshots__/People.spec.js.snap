// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People matches snapshot 1`] = `
<PeopleList
  error={null}
  loading={false}
  people={
    Array [
      Object {
        "id": 0,
        "name": "Ankur",
        "role": "CEO",
      },
      Object {
        "id": 1,
        "name": "Aliya",
        "role": "Developer",
      },
      Object {
        "id": 2,
        "name": "Bassam",
        "role": "Developer",
      },
    ]
  }
/>
`;

exports[`People renders a error message when request fails 1`] = `
<StateMessage
  className="error"
  state="error"
>
  There was an error loading list of People.
</StateMessage>
`;

exports[`People renders a loading message when requesting data 1`] = `
<StateMessage
  className="loading"
  state="loading"
>
  Loading the list of People...
</StateMessage>
`;
